src/tye/InitHost.cs

```diff
@@ -105,5 +110,49 @@ private static void ThrowIfTyeFilePresent(FileInfo? path, string yml)
                 throw new CommandException($"File '{tyeYaml}' already exists. Use --force to override the {yml} file if desired.");
             }
         }
+
+        private static void AddToSolutionItems(FileInfo? path)
+        {
+            var slnFiles = Directory.GetFiles(path!.DirectoryName, "*.sln");
+            if (slnFiles.Length == 1)
+            {
+                var sln = slnFiles[0];
+
+                var lines = File.ReadAllLines(sln).ToList<string>();
+                var insertAt = 0;
+                var exists = false;
+
+                lines.ForEach(l =>
+                {
+                    if (l.IndexOf("(SolutionItems)") > 0)
+                        exists = true;
+
+                    if (!exists)
+                        insertAt++;
+                });
+
+                var insertItem = string.Empty;
+                if (exists)
+                {
+                    insertItem = "        tye.yaml = tye.yaml";
+                    lines.Insert(insertAt + 1, insertItem);
+                }
+                else
+                {
+                    var fileGuid = Guid.NewGuid().ToString("B").ToUpper();
+
+                    insertItem = @$"Project(""{fileGuid}"") = ""Solution Items"", ""Solution Items"", ""{fileGuid}""
```

Comment:
No this is still wrong, the two Guids have different purposes and should not be the same. Again, how are you verifying this change? I think if you opened the sln file, there will be a load error.

Reply:
The `SolutionItems` is a good call out here. Testing process has been using the simple microservices.sln example from the docs. Opening the sln in VS works fine. There is no loading error.

Ideally, there should be `dotnet sln add` functionality in the dotnet CLI to support this beyond **proj files.

Reply:
When I test this locally, I see
![image](https://user-images.githubusercontent.com/2030323/87361085-4b3aee00-c520-11ea-8cea-a8359f92801f.png)

It's because the project type Guid is invalid if you just use a randomly generated Guid.

Reply:
And I agree, maybe file an issue here https://github.com/dotnet/sdk/issues for `dotnet sln` to have the ability to add solution file? Also this is why I mentioned that it would be better to use the Microsoft.DotNet.Cli.Sln.Internal library if we plan to update our TFM.

Reply:
Resolved the Guid issue. There is a section Guid, project references should have the same in the first and then an identifier for the second. Solution items are their own section. Taking a dependency on on dotnet cli commands would be preferred over Microsoft.DotNet.Cli.Sln.Internal in the future. 

Added issue :https://github.com/dotnet/sdk/issues/12454